building/build-debs/homeworld-admin-tools/src/access.py

```diff
@@ -66,11 +66,17 @@ def access_ssh(add_to_agent=False):
     print("===== ^ CERTIFICATE DETAILS ^ =====")
     if add_to_agent:
         # TODO: clear old identities
-        if subprocess.call(["ssh-add", "--", keypath]) != 0:
-            print("ssh-add returned non-zero exit code. do you have a ssh-agent?")
-            print("or, perhaps your agent is broken; consider killing it and launching a new one.")
-            command.fail("*** ssh-add failed! ***")
-
+        try:
+            ssh_add_output = subprocess.check_output(["ssh-add", "--", keypath], stderr=subprocess.STDOUT)
+            # if the user is using gnome, gnome-keyring might
+            # masquerade as ssh-agent and provide a zero exit
+            # code despite failing to add the certificate
+            if "add failed".encode('utf-8') in ssh_add_output:
```

Comment:
Do we need to specify the codec? I thought it defaulted to utf-8.

Reply:
That's correct. Thanks!